src/components/MultiHopTrade/components/FiatRamps/FiatRampQuoteCard.tsx

```diff
@@ -1,63 +1,182 @@
+        .toString()
+    }, [quote.amount, buyAssetMarketData])
+
+    const buyAmountCryptoPrecision = useMemo(() => {
+      if (direction === FiatRampAction.Buy) {
+        return bnOrZero(quote.amount)
+          .div(sellAssetMarketData?.price ?? 0)
+          .toString()
+      }
+
+      const usdRate = fiatMarketData[buyFiatCurrency?.code as SupportedFiatCurrencies]?.price ?? 0
+
+      if (buyFiatCurrency?.code === selectedUserCurrency || !usdRate)
+        return bnOrZero(quote.amount)
+          .div(sellAssetMarketData?.price ?? 0)
+          .toFixed(sellAsset?.precision ?? 0)
+
+      return bnOrZero(quote.amount)
+        .div(usdRate)
+        .div(sellAssetMarketData?.price ?? 0)
+        .toFixed(sellAsset?.precision ?? 0)
+    }, [
+      quote.amount,
+      sellAssetMarketData,
+      fiatMarketData,
+      direction,
+      buyFiatCurrency,
+      sellAsset,
+      selectedUserCurrency,
+    ])
+
     const handleQuoteSelection = useCallback(() => {
-      // TODO: Handle quote selection for fiat ramp
-      console.log('Selected fiat ramp quote:', quote)
-      // For mobile, if the user click on a quote, it should go back
-      // even thought I didn't really take care of mobile yet
+      dispatch(tradeRampInput.actions.setSelectedFiatRampQuote(quote))
+
       onBack && onBack()
-    }, [quote, onBack])
+    }, [quote, onBack, dispatch])
 
     const providerIcon = useMemo(() => {
       return <AssetIcon src={quote.providerLogo} />
     }, [quote.providerLogo])
 
+    const fiatAmountDisplay = useMemo(() => {
+      if (!fiatCurrency || !quote.rate) return '0'
+
+      if (direction === FiatRampAction.Buy) {
+        return buyAmountUserCurrency
+      }
+
+      return quote.amount
+    }, [fiatCurrency, quote.rate, quote.amount, direction, buyAmountUserCurrency])
+
     const headerContent = useMemo(() => {
       return (
         <Flex justifyContent='space-between' alignItems='center' flexGrow={1}>
           <Box ml='auto'>
             <TradeQuoteBadges
               isBestRate={isBestRate}
-              isFastest={isFastest}
               quoteDisplayOption={QuoteDisplayOption.Basic}
             />
           </Box>
         </Flex>
       )
-    }, [isBestRate, isFastest])
+    }, [isBestRate])
+
+    const amounts = useMemo(() => {
+      if (direction === FiatRampAction.Buy) {
+        return (
+          <>
+            <Amount.Crypto
+              value={quote.amount}
+              symbol={buyAsset.symbol}
+              fontSize='lg'
+              fontWeight='bold'
+              color='text.base'
+              maximumFractionDigits={8}
+              omitDecimalTrailingZeros={true}
+            />
+            <Amount.Fiat value={fiatAmountDisplay} fontSize='sm' color='text.subtle' prefix='≈' />
+          </>
+        )
+      }
+
+      return (
+        <>
+          <Amount.Fiat
+            value={fiatAmountDisplay}
+            fiatType={buyFiatCurrency?.code}
```

Comment:
looks incorrect for sell side?

Reply:
Not sure to get this one? seems correct at runtime